File: src/Types/PreBlock.hs
{-# LANGUAGE DeriveGeneric #-}

module Types.PreBlock where

import GHC.Generics (Generic)
import Data.Aeson (ToJSON, FromJSON)

import Types.Transaction

data PreBlock = PreBlock
  { index :: Int
  , timestamp :: Int
  , transactions :: [Transaction]
  , previousHash :: String
  , nonce :: Int
  } deriving (Show, Eq, Generic)

instance ToJSON PreBlock
instance FromJSON PreBlock

File: src/main/kotlin/block/PreBlock.kt
package block

import transactions.Transaction

data class PreBlock(
    private val index: Int,
    private val timestamp: Long,
    private val transactions: List<Transaction>,
    private val previousHash: String,
    private val nonce: Long,
) {
    fun getIndex(): Int = index
    fun getTimestamp(): Long = timestamp
    fun getTransactions(): List<Transaction> = transactions
    fun getPreviousHash(): String = previousHash
    fun getNonce(): Long = nonce
}
